app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt

```diff
@@ -0,0 +1,84 @@
+package org.oppia.app.parser
+
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import android.text.Spannable
+import android.widget.TextView
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import com.google.common.truth.Truth.assertThat
+import org.hamcrest.Matchers.not
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.testing.HtmlParserTestActivity
+import org.oppia.util.parser.HtmlParser
+import javax.inject.Inject
+
+/** Tests for [HtmlParser]. */
+@RunWith(AndroidJUnit4::class)
+class HtmlParserTest {
```

Comment:
Please add a test to verify that image loading works. If you swap out the ImageLoader for a mock of it, you should be able to verify that it gets called.

Reply:
@BenHenning I have added testcases but it fails I have tried this approach app/src/sharedTest/java/org/oppia/app/parser/UrlImageParserTest.kt.

I tried testing with `place_holders` but I cant add place_holder in `utility` module so here I am testing if drawable returned is null or not.

Reply:
Can you please be specific on what is failing? Without more details I can't help debug.

Also when glancing at that test I noticed that you're replacing the injected image loader with a mock. We need to bind a mock of ImageLoader to the Dagger graph instead of pulling in the Glide version (we shouldn't use the GlideImageLoader in tests) in order for the prod code to call into the mock that the test is setting up.

Reply:
It fails to load image. It is returning null for drawable.

I tried changing implementation using bind, but it was throwing me error https://github.com/oppia/oppia-android/pull/205#discussion_r337332902

Reply:
As discussed via Chat and email, this isn't possible to test without fully resolving #59.